### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by RWUBAKWANAYO**
# In this project we:

- [ ]  Create a class `Classroom` with the following:
- `@label` instance variable, should be initialized in the constructor.
- Setter and getter for `@label` (remember about `attr_accessor`).

- [ ]  Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student` by :
- Create the `has-many` side (a classroom has many students).
- Create the `belongs-to` side (a student belongs to a classroom).
- Make sure that when adding a student to a classroom it also sets the classroom for the student.
- Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- [ ] Create a class `Book` with the following:
- `@title` and `@author` instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about `attr_accessor`).

- [ ] Create a class `Rental` with the following:
- `@date` instance variable, should be initialized in the constructor.
- Setter and getter for `@date` (remember about `attr_accessor`).

- [ ] Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental` by :

- Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
- Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
-  Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
- Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
-  Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (omar-muhamad) [inline book.rb]: - [ ] Please add getters and setters for @rentals `attr_accessor` instead of getters only `attr_reader`.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (omar-muhamad) [inline rental.rb]: - [ ] Same here, Kindly add getters and setters for `@person` and `@book` instead of getters only.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @RWUBAKWANAYO,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @RWUBAKWANAYO,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by RWUBAKWANAYO** [comment]
merge changes to the dev branch

**Pull request merged**